packages/network-controller/src/NetworkController.ts

```diff
@@ -938,16 +1007,24 @@ export class NetworkController extends BaseControllerV2<
   /**
    * Fetches the latest block for the network.
    *
+   * @param networkClientId - The networkClientId to fetch the correct provider against which to check the latest block. Defaults to the selectedNetworkClientId.
    * @returns A promise that either resolves to the block header or null if
    * there is no latest block, or rejects with an error.
    */
-  #getLatestBlock(): Promise<Block> {
-    return new Promise((resolve, reject) => {
-      if (!this.#ethQuery) {
-        throw new Error('Provider has not been initialized');
-      }
+  #getLatestBlock(networkClientId?: NetworkClientId): Promise<Block> {
```

Comment:
Since this is a private method, we control how it's called. What do you think about having this method always take a network client ID to simplify its signature?